provider/azure/azure_discover.go

```diff
@@ -40,14 +50,22 @@ func (p *Provider) Help() string {
    resource_group:    The name of the resource group to filter on
    vm_scale_set:      The name of the virtual machine scale set to filter on
 
-   When using tags the only permission needed is the 'ListAll' method for
-   'NetworkInterfaces'. When using Virtual Machine Scale Sets the only Role
-   Action needed is 'Microsoft.Compute/virtualMachineScaleSets/*/read'.
+   When using tags the only permission needed is Microsoft.Network/networkInterfaces/*
+
+	 When using Virtual Machine Scale Sets the only role action needed is Microsoft.Compute/virtualMachineScaleSets/*/read.
 
    It is recommended you make a dedicated key used only for auto-joining.
 `
 }
 
+// argsOrEnv allows you to pick an environmental variable for a setting if the arg is not set
+func argsOrEnv(args map[string]string, key, env string) string {
+	if value := args[key]; value != "" {
```

Comment:
you could also check the second returned value: `if value, ok := args[key]; ok {`

Reply:
Fixed! 👍